src/Microsoft.SqlTools.ServiceLayer/EditData/Session.cs

```diff
@@ -110,6 +114,29 @@ public long CreateRow()
         }
 
         /// <summary>
+        /// Commits the edits in the cache to the database and then to the associated result set of
+        /// this edit session. This is launched asynchronously.
+        /// </summary>
+        /// <param name="connection">The connection to use for executing the query</param>
+        /// <param name="successHandler">Callback to perform when the commit process has finished</param>
+        /// <param name="errorHandler">Callback to perform if the commit process has failed at some point</param>
+        public void CommitEdits(DbConnection connection, Func<Task> successHandler, Func<Exception, Task> errorHandler)
```

Comment:
Yeah, I think I could do it as you suggested. The downsides to it are mostly that I'd end up having code nested inside the construction of a Task. While not terrible, I also followed this pattern for query execution, so I think I'll just stay with it as-is.